lib/storage/src/rbac/mod.rs

```diff
@@ -23,13 +23,15 @@ pub enum Access {
 #[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
 pub struct CollectionAccessList(pub Vec<CollectionAccess>);
 
+pub struct ExistingCollections {
+    inner: HashSet<String>,
+}
+
 #[derive(Serialize, Deserialize, Validate, PartialEq, Clone, Debug)]
+#[validate(context = ExistingCollections, mutable)]
 pub struct CollectionAccess {
     /// Collection names that are allowed to be accessed
-    #[validate(custom(
-        function = "validate_unique_collections",
-        arg = "&'v_a mut HashSet<String>"
-    ))]
+    #[validate(custom(function = "validate_unique_collections", use_context))]
```

Comment:
the new macro refused to work with `HashSet<String>` and forced me to create a new struct.

See context in https://github.com/Keats/validator?tab=readme-ov-file#custom

Reply:
:disappointed: 

Since this struct doesn't leak to other places I'm totally fine with it.